zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SearchEngineClient.java

```diff
@@ -20,20 +21,9 @@ void createIndexTemplate(
       final Boolean create);
 
   /**
-   * The {@code propertiesJson} should have a root properties field, for example:
-   *
-   * <pre>{@code
-   * {
-   *   "properties" : {
-   *     "field" : {
-   *       "type" : "keyword"
-   *     }
-   *   }
-   * }
-   * }</pre>
-   *
    * @param indexDescriptor Representing index of which to update the mappings
-   * @param propertiesJson New properties to be appended to the index
+   * @param newProperties New properties to be appended to the index
    */
-  void putMapping(final IndexDescriptor indexDescriptor, final String propertiesJson);
+  void putMapping(
+      final IndexDescriptor indexDescriptor, final Set<IndexMappingProperty> newProperties);
```

Comment:
:wrench: Why don't we just use `Map<String, Object>` instead of `Set<IndexMappingProperty`. To serialize it to json, you are building a map anyway.

Reply:
I was trying to keep it as compatible with the current operate code as possible to save time making things work. The `IndexSchemaValidator` in the `validateIndexMappings()` function returns for every index descriptor a `Set<IndexMappingProperty>` detailing any additional properties to add. Doing it this way requires no marshalling of different types although I agree with your point regarding the serialisation. 

In conclusion the way the difference is calculated as per `IndexMappingDifference` will return a `Set<IndexMappingProperty>` at some point in time this will need to be serialised to json or converted to an els/opensearch data structure to be passed to the request, and in the request will hide implementation details to users of the client.

Reply:
:+1: Since this do not affect the actual Operate code, I think it is ok to change it. But let's leave it like this and consider refactoring later when implementing the validation.